index.js

```diff
@@ -0,0 +1,169 @@
+/*
+
+review copy for the los angeles software craftsmanship community
+
+original code from the request project is here:
+
+    https://github.com/request/request/blob/master/index.js
+
+*/
+
+
+
+
+// Copyright 2010-2012 Mikeal Rogers
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+'use strict'
+
+var extend                = require('extend')
+  , cookies               = require('./lib/cookies')
+  , helpers               = require('./lib/helpers')
+
+var isFunction            = helpers.isFunction
+  , paramsHaveRequestBody = helpers.paramsHaveRequestBody
+
+
```

Comment:
I immediately assumed this was too many parameters for the following function, but is it? Passing callbacks into a function is something I don't think was really addressed in Clean Code, or I've forgotten... Also, I do think this function is too enigmatic -- I really can't understand what it does by its name, or even reading the code for that matter